var/spack/repos/builtin/packages/hsa-rocr-dev/package.py

```diff
@@ -49,7 +53,18 @@ def cmake_args(self):
         libelf_include = self.spec['libelf'].prefix.include.libelf
         args = ['-DLIBELF_INCLUDE_DIRS=%s' % libelf_include]
 
-        if '@3.7.0:' in self.spec:
+        if '+shared' in self.spec:
+            args.append('-DBUILD_SHARED_LIBS=ON')
+        else:
+            args.append('-DBUILD_SHARED_LIBS=OFF')
```

Comment:
Can you use `self.define_from_variant(.., ..)` where applicable to reduce code bloat?